roofit/roofitcore/inc/UniqueId.h

```diff
@@ -0,0 +1,73 @@
+// Author: Jonas Rembser, CERN, Jun 2021
+
+#ifndef roofit_roofitcore_UniqueId_h
+#define roofit_roofitcore_UniqueId_h
+
+/// A UniqueId can be added as a class member to enhance any class with a
+/// unique identifier for each instantiated object.
+///
+/// Example:
+/// ~~~{.cpp}
+/// class MyClass {
+///
+/// public:
+///    /// Return unique ID by reference.
+///    /// Please always use the name `uniqueId` for the getter.
+///    UniqueId<MyClass> const& uniqueId() const { return _uniqueId; }
+///
+/// private:
+///    const UniqueId<MyClass> _uniqueId; //! should be non-persistent
+///
+/// };
+/// ~~~
+template <class Class>
+struct UniqueId {
+
+public:
+
+    using Value_t = unsigned long;
```

Comment:
Indentation error?

Reply:
Every time a completely new file is introduced, I thought it was a good opportunity to adopt the official clang-format [ROOT style](https://github.com/root-project/root/blob/master/.clang-format). That style indents with three spaces. I did this first with the new RooCrystalBall. I also ask other contributors to format their sources accordingly. You think that's not a good idea?

Reply:
No, I would write a clang-format style for RooFit.

And this one is actually indented with 4 spaces. :-)